src/modules/Game.class.js

```diff
@@ -1,68 +1,328 @@
+
+    this.board[randomRow][randomCol] = Math.random() < 0.9 ? 2 : 4;
+
+    this.updateBoardDisplay();
+
+    const newTile = document.querySelector(`
+      .field-row:nth-child(${randomRow + 1})
+      .field-cell:nth-child(${randomCol + 1})
+    `);
+
+    newTile.classList.add('tile-new');
+
+    setTimeout(() => {
+      newTile.classList.remove('tile-new');
+    }, 300);
+  }
+
+  moveLeft() {
+    const { board } = this;
+    const oldBoard = this.saveBoardState();
+    const newBoard = Array.from({ length: 4 }, () => Array(4).fill(0));
+    let score = 0;
+
+    for (let i = 0; i < 4; i++) {
+      let position = 0;
+
+      for (let j = 0; j < 4; j++) {
+        if (board[i][j] !== 0) {
+          if (newBoard[i][position] === board[i][j]) {
+            newBoard[i][position] *= 2;
+            score += newBoard[i][position];
+
+            position++;
+          } else if (newBoard[i][position] === 0) {
+            newBoard[i][position] = board[i][j];
+          } else {
+            position++;
+            newBoard[i][position] = board[i][j];
+          }
+        }
+      }
+    }
+
+    this.score += score;
+
+    this.board = newBoard;
+
+    if (this.hasBoardChanged(oldBoard, this.board)) {
+      this.addRandomTitle();
+    }
+
+    this.updateBoardDisplay();
+    this.updateScoreDisplay();
+
+    return this.score;
+  }
+
+  moveRight() {
+    const { board } = this;
+    const newBoard = Array.from({ length: 4 }, () => Array(4).fill(0));
+    let score = 0;
+
+    for (let i = 0; i < 4; i++) {
+      let position = 3;
+
+      for (let j = 3; j >= 0; j--) {
+        if (board[i][j] !== 0) {
+          if (newBoard[i][position] === board[i][j]) {
+            newBoard[i][position] *= 2;
+            score += newBoard[i][position];
+            position--;
+          } else if (newBoard[i][position] === 0) {
+            newBoard[i][position] = board[i][j];
+          } else {
+            position--;
+            newBoard[i][position] = board[i][j];
+          }
+        }
+      }
+    }
+
+    this.score += score;
+    this.board = newBoard;
+
+    this.addRandomTitle();
+    this.updateScoreDisplay();
+
+    return this.score;
+  }
+
+  moveUp() {
+    const { board } = this;
+    const newBoard = Array.from({ length: 4 }, () => Array(4).fill(0));
+    let score = 0;
+
+    for (let j = 0; j < 4; j++) {
+      let position = 0;
+
+      for (let i = 0; i < 4; i++) {
+        if (board[i][j] !== 0) {
+          if (newBoard[position][j] === board[i][j]) {
+            newBoard[position][j] *= 2;
+            score += newBoard[position][j];
+            position++;
+          } else if (newBoard[position][j] === 0) {
+            newBoard[position][j] = board[i][j];
+          } else {
+            position++;
+            newBoard[position][j] = board[i][j];
+          }
+        }
+      }
+    }
+
+    this.score += score;
+    this.board = newBoard;
+
+    this.addRandomTitle();
+    this.updateScoreDisplay();
```

Comment:
looks like, you can just move this logic here, after the action
<img width="471" alt="image" src="https://github.com/user-attachments/assets/3332f222-ef34-46fc-95f1-6f5988c08935">

and remove from all actions functions